test/integration/net_test.go

```diff
@@ -195,6 +183,23 @@ func TestNetworkPlugins(t *testing.T) {
 	})
 }
 
+// verifyHairpinMode makes sure the hairpinning (https://en.wikipedia.org/wiki/Hairpinning) is correctly configured for given CNI
```

Comment:
add more details, what exactly are we testing ? consider wrtiting this comment for someone who didnt know what is hairpin before